File: VB/E1885/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports System.Diagnostics
Imports DevExpress.XtraGrid.Views.Layout.Events
Imports DevExpress.XtraGrid.Views.Layout

Namespace E1885
	Partial Public Class Form1
		Inherits XtraForm
		Private rnd As New Random()
		Private cache As New Dictionary(Of String, Timer)()
		Private regular As Font
		Private modified As Font

		Public Sub New()
			InitializeComponent()
			CreateData()
			regular = layoutView1.Appearance.FieldValue.Font
			modified = New Font(layoutView1.Appearance.FieldValue.Font, FontStyle.Bold)
			timer1.Start()
		End Sub

		Private Sub CreateData()
			For i As Integer = 0 To 9
				dataTable1.Rows.Add(rnd.Next(100).ToString(), rnd.Next(100).ToString())
			Next i
		End Sub

		Private Sub OnBindingSourceListChanged(ByVal sender As Object, ByVal e As ListChangedEventArgs) Handles bindingSource1.ListChanged
			If e.ListChangedType = ListChangedType.ItemChanged Then
				Dim key As String = String.Format("{0}|{1}", e.NewIndex, e.PropertyDescriptor.Name)
				Dim timer As Timer
				If cache.ContainsKey(key) Then
					timer = cache(key)
					timer.Stop()
					timer.Start()
				Else
					timer = New Timer()
					timer.Interval = 1000
					timer.Tag = key
					AddHandler timer.Tick, AddressOf OnCacheExpired
					timer.Start()
					cache.Add(key, timer)
				End If
			End If
		End Sub

		Private Sub OnTimerClick(ByVal sender As Object, ByVal e As EventArgs) Handles timer1.Tick
			dataTable1.Rows(rnd.Next(dataTable1.Rows.Count))(rnd.Next(2)) = rnd.Next(100).ToString()
		End Sub

		Private Sub OnCacheExpired(ByVal sender As Object, ByVal e As EventArgs)
			Dim timer As Timer = CType(sender, Timer)
			Dim key As String = CStr(timer.Tag)
			cache.Remove(key)
			timer.Dispose()
			Dim positionData() As String = key.Split("|"c)
			Dim rowHandle As Integer = layoutView1.GetRowHandle(Integer.Parse(positionData(0)))
			layoutView1.RefreshRow(rowHandle)
		End Sub

		Private Sub OnLayoutViewCustomFieldValueStyle(ByVal sender As Object, ByVal e As LayoutViewFieldValueStyleEventArgs) Handles layoutView1.CustomFieldValueStyle
			Dim view As LayoutView = CType(sender, LayoutView)
			Dim sourceIndex As Integer = view.GetDataSourceRowIndex(e.RowHandle)
			Dim key As String = String.Format("{0}|{1}", sourceIndex, e.Column.FieldName)
			If cache.ContainsKey(key) Then
				e.Appearance.Font = modified
			Else
				e.Appearance.Font = regular
			End If
		End Sub

		Private Overloads Sub OnFormClosed(ByVal sender As Object, ByVal e As FormClosedEventArgs) Handles MyBase.FormClosed
			modified.Dispose()
			modified = Nothing
		End Sub
	End Class
End Namespace

File: CS/E1885/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using System.Diagnostics;
using DevExpress.XtraGrid.Views.Layout.Events;
using DevExpress.XtraGrid.Views.Layout;

namespace E1885 {
    public partial class Form1 : XtraForm {
        private Random rnd = new Random();
        private Dictionary<string, Timer> cache = new Dictionary<string, Timer>();
        private Font regular;
        private Font modified;

        public Form1() {
            InitializeComponent();
            CreateData();
            regular = layoutView1.Appearance.FieldValue.Font;
            modified = new Font(layoutView1.Appearance.FieldValue.Font, FontStyle.Bold);
            timer1.Start();
        }

        private void CreateData() {
            for (int i = 0; i < 10; i++)
                dataTable1.Rows.Add(rnd.Next(100).ToString(), rnd.Next(100).ToString());
        }

        private void OnBindingSourceListChanged(object sender, ListChangedEventArgs e) {
            if (e.ListChangedType == ListChangedType.ItemChanged) {
                string key = string.Format("{0}|{1}", e.NewIndex, e.PropertyDescriptor.Name);
                Timer timer;
                if (cache.ContainsKey(key)) {
                    timer = cache[key];
                    timer.Stop();
                    timer.Start();
                } else {
                    timer = new Timer();
                    timer.Interval = 1000;
                    timer.Tag = key;
                    timer.Tick += new EventHandler(OnCacheExpired);
                    timer.Start();
                    cache.Add(key, timer);
                }
            }
        }

        private void OnTimerClick(object sender, EventArgs e) {
            dataTable1.Rows[rnd.Next(dataTable1.Rows.Count)][rnd.Next(2)] = rnd.Next(100).ToString();
        }

        private void OnCacheExpired(object sender, EventArgs e) {
            Timer timer = (Timer)sender;
            string key = (string)timer.Tag;
            cache.Remove(key);
            timer.Dispose();
            string[] positionData = key.Split('|');
            int rowHandle = layoutView1.GetRowHandle(int.Parse(positionData[0]));
            layoutView1.RefreshRow(rowHandle);
        }

        private void OnLayoutViewCustomFieldValueStyle(object sender, LayoutViewFieldValueStyleEventArgs e) {
            LayoutView view = (LayoutView)sender;
            int sourceIndex = view.GetDataSourceRowIndex(e.RowHandle);
            string key = string.Format("{0}|{1}", sourceIndex, e.Column.FieldName);
            e.Appearance.Font = cache.ContainsKey(key) ? modified : regular;
        }

        private void OnFormClosed(object sender, FormClosedEventArgs e) {
            modified.Dispose();
            modified = null;
        }
    }
}